src/components/frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionSettings/utils.tsx

```diff
@@ -31,37 +33,180 @@ import {
   getGlobalThresholdType,
 } from './MultiSigSettings/utils.ts';
 
-export const getFormSuccessFn =
-  <T extends FieldValues>({
-    setWaitingForActionConfirmation,
-    extensionData,
-    refetchExtensionData,
-    setActiveTab,
-  }: {
-    setWaitingForActionConfirmation: SetStateFn;
-    setActiveTab: (tabId: ExtensionDetailsPageTabId) => void;
-    extensionData: AnyExtensionData;
-    refetchExtensionData: RefetchExtensionDataFn;
-  }): OnSuccess<T> =>
-  async (fieldValues, { reset }) => {
-    setWaitingForActionConfirmation(true);
-    const isSaveChanges =
-      extensionData &&
-      isInstalledExtensionData(extensionData) &&
-      (extensionData.isInitialized || extensionData.configurable);
-
-    try {
-      if (!isSaveChanges) {
-        await waitForDbAfterExtensionAction({
-          method: ExtensionMethods.ENABLE,
-          refetchExtensionData,
-        });
+const getInitializationDefaultValues = (
+  initializationParams: ExtensionInitParam[],
+) => {
+  return initializationParams.reduce<
+    Record<string, string | number | undefined>
+  >((initialValues, param) => {
+    return {
+      ...initialValues,
+      [param.paramName]: param.defaultValue,
+    };
+  }, {});
+};
 
-        reset();
+export const getExtensionSettingsDefaultValues = (
+  extensionData: AnyExtensionData,
+): object => {
+  const { initializationParams = [] } = extensionData;
+  const defaultValues = getInitializationDefaultValues(initializationParams);
+
+  if (!isInstalledExtensionData(extensionData)) {
+    return defaultValues;
+  }
 
+  switch (extensionData.extensionId) {
+    case Extension.StakedExpenditure: {
+      return {
+        stakeFraction: extensionData?.params?.stakedExpenditure?.stakeFraction
+          ? convertFractionToEth(
+              extensionData.params.stakedExpenditure.stakeFraction,
+            )
+          : String(defaultValues.stakeFraction),
+      };
+    }
+    case Extension.MultisigPermissions: {
+      const { colonyThreshold: globalThreshold } =
+        extensionData.params?.multiSig ?? {};
+
+      return {
+        thresholdType:
+          globalThreshold !== undefined
+            ? getGlobalThresholdType(globalThreshold)
+            : MultiSigThresholdType.MAJORITY_APPROVAL,
+        globalThreshold: globalThreshold ?? 0,
+        domainThresholds: [],
+      } satisfies MultiSigSettingsFormValues;
+    }
+    default: {
+      return defaultValues;
+    }
+  }
+};
```

Comment:
This is just moving an existing function up to please the liner. Below here are the actual changes.